sql/core/src/main/scala/org/apache/spark/sql/catalyst/SQLBuilder.scala

```diff
@@ -512,8 +512,12 @@ class SQLBuilder(logicalPlan: LogicalPlan) extends Logging {
           ScalarSubquery(rewrite, Seq.empty, exprId)
 
         case PredicateSubquery(query, conditions, false, exprId) =>
-          val plan = Project(Seq(Alias(Literal(1), "1")()),
-            Filter(conditions.reduce(And), addSubqueryIfNeeded(query)))
+          val plan = if (conditions.isEmpty) {
+            addSubqueryIfNeeded(query)
```

Comment:
Minor: it is a little more readable if we move the `addSubqueryIfNeeded(query)` out of the if...else... and into a separate variable.